x-pack/plugins/apm/public/components/app/service_logs/index.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLoadingSpinner, EuiEmptyPrompt } from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
+import { LogStream } from '../../../../../infra/public';
+
+export function ServiceLogs() {
+  const { serviceName } = useApmServiceContext();
+  const {
+    urlParams: { environment, kuery, start, end },
+  } = useUrlParams();
+
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      if (start && end) {
+        return callApmApi({
+          endpoint: 'GET /api/apm/services/{serviceName}/infrastructure',
+          params: {
+            path: { serviceName },
+            query: {
+              environment,
+              kuery,
+              start,
+              end,
+            },
+          },
+        });
+      }
+    },
+    [environment, kuery, serviceName, start, end]
+  );
+
+  const noInfrastructureData = useMemo(() => {
+    const containerIdsLength =
+      data?.serviceInfrastructure?.containerIds?.length ?? 0;
+    const hostNamesLength = data?.serviceInfrastructure?.hostNames?.length ?? 0;
+    const podNamesLength = data?.serviceInfrastructure?.podNames?.length ?? 0;
+    return (
+      containerIdsLength === 0 && hostNamesLength === 0 && podNamesLength === 0
+    );
```

Comment:
Perhaps a bit cleaner
```suggestion    
  return (
    isEmpty(data?.serviceInfrastructure?.containerIds) &&
    isEmpty(data?.serviceInfrastructure?.hostNames) &&
    isEmpty(data?.serviceInfrastructure?.podNames)
  );
```